lib/ExpensiMark.js

```diff
@@ -49,6 +49,39 @@ export default class ExpensiMark {
                 replacement: '$1<code>$2</code>$3',
             },
 
+            /**
+             * Apply the hereMention first because the string @here is still a valid mention for the userMention regex.
+             * This ensures that the hereMention is always considered first, even if it is followed by a valid userMention.
+            */
+            {
+                name: 'hereMentions',
+                regex: /((&#x60;|<code>|<pre>)\s*?)?[`.a-zA-Z]?@here\b/gm,
+                replacement: (match) => {
+                    if (!Str.isValidMention(match)) {
+                        return match;
+                    }
+                    return `<mention-here>${match}</mention-here>`;
+                },
+            },
+
+            /**
+             * This regex matches a valid user mention in a string.
+             * A user mention is a string that starts with the '@' symbol and is followed by a valid user's primary login
+             *
+             * Note: currently we are only allowing mentions in a format of @+19728974297@expensify.sms and @username@example.com
+             * The username can contain any combination of alphanumeric letters, numbers, and underscores
+             */
+            {
+                name: 'userMentions',
+                regex: new RegExp(`((&#x60;|<code>|<pre>)\\s*?)?[\`.a-zA-Z]?@+${CONST.REG_EXP.EMAIL_PART}`, 'gm'),
```

Comment:
👋 Dropping a note about this causing a regression in https://github.com/Expensify/App/issues/18857. Essentially just missed a case where mention is inside of a link. More details are [here](https://github.com/Expensify/App/issues/18857#issuecomment-1546670288)